### PR TITLE
[WIP] migrate to GitHub Actions (#616)

### PR DESCRIPTION
**Pull request by tomkralidis**
Per #616

### CODE REVIEW

**Comment by tomkralidis** [comment]
@kalxas @justb4 @jodygarnett @francbartoli 

FYI GitHub Actions setup passing so far.  Here's what is outstanding/not ported from the Travis setup in this PR:

- loading PostgreSQL/PostGIS data
- properly getting the GDAL Python bindings properly installed
- properly getting the pyspatialite dependencies installed
- running any tests that depend on the above
- anything Docker
- building Debian package

I've listed out each test in https://github.com/tomkralidis/pygeoapi/blob/issue-616/.github/workflows/main.yml so as to add/remove accordingly as part of testing.  Once all tests pass, then this step will simply be run at `pytest`.

Any help is appreciated to help get our tests working again.  Having said this, we may want to consider merging even with the abovementioned gaps given Travis testing is blocked by limits atm.

**Comment by kalxas** [comment]
This looks great. Having this working in GH Actions is better that a more complete Travis list of test that will shut down soon.

**Review by kalxas — APPROVED** [review]

**Review by francbartoli — APPROVED** [review]
I’m inclined to merge as it is since we are not able to have any CI now. Then we can add the rest of the pieces missing from the travis pipeline

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # pygeoapi
 
-[![Build Status](https://travis-ci.org/geopython/pygeoapi.png)](https://travis-ci.org/geopython/pygeoapi)
+[![Build Status](https://github.com/geopython/pygeoapi/workflows/build%20%E2%9A%99%EF%B8%8F/badge.svg)](https://github.com/geopython/pygeoapi/actions)
```
Comment (justb4) [inline README.md:3]: think the badge string should be
 
[Build Status](https://github.com/geopython/pygeoapi/actions/workflows/main.yml)

**Review by justb4 — APPROVED** [review]
I think good to go as first step. Then you have something working.
Next we can fix GDAL (and uncomment from requirements-provider.txt) and other deps. Beste to look into the pygeoapi Dockerfile, where e.g. the deb packages `python3-gdal python3-psycopg2 python3-xarray` are installed.

Then we can also add the Docker build, test and push. As @jodygarnett indicated can be separate file. And maybe more selective (maybe only on push trigger).
I have some examples from other projects like:
https://github.com/justb4/docker-awstats/blob/master/.github/workflows/main.yml 
Also much faster than building on DockerHub.

**Pull request merged**